CRM/Event/Form/Registration/Confirm.php

```diff
@@ -456,6 +457,15 @@ public function postProcess() {
     $participantCount = [];
     $totalTaxAmount = 0;
 
+    if ($this->isShowPaymentOnConfirm()) {
+      // We assign this here so we don't assign it multiple times in the foreach below
+      $submitValues = $this->getSubmitValues();
```

Comment:
@mattwire `getSubmittedValues()` is the safe method  -or `$this->_paymentProcessorIDs = [$this->getSubmittedValues('payment_processor_id')];` - it handles money formatting & checks the field is present - the other basically just gets `_POST`

Reply:
How is it getting into submitValues but not submittedValues - that would happen if the field were not added to the form but somehow posting anyway? Or is it being added to the `Confirm` form but this function is too 'Main-biased' https://github.com/civicrm/civicrm-core/blob/8dd4dec4e0ee44d53ebbd1681a21d16431061e6d/CRM/Contribute/Form/ContributionBase.php#L1402

Ideally we would add a function `getPaymentProcessorID()` as that is the pattern we are now adding - if that is there then including `CRM_Financial_Form_PaymentProcessorFormTrait` would give us `getPaymentProcessorValue()` too

Reply:
@mattwire where are the fields added to the form?

Reply:
@mattwire ah ok - you are not calling that `buildQuickForm()` when `isSubmitted()` is true - which would prevent the fields being added to the quickForm - hence `$this->getSubmittedValue('payment_processor_id')` doesn't find the field on the form & so filters it out

Reply:
@mattwire it might be worth adding a variant of https://github.com/civicrm/civicrm-core/blob/12cd7925fd9f752173d3d39781fe29e25814e298/CRM/Contribute/Form/ContributionBase.php#L1353-L1375 to the Event form flow now - because if it gets added later other people (like me) might not recall that values can now be submitted on the Confirm page & that would cause a regression that affects you.

It could be much like that one but just also merge in Confirm fields

Reply:
I didn't add a `getSubmittedValue` function. I'm not sure it makes sense here as there are too many forms involved that can submit values (eg. "Register", "additionalparticipant", "Confirm") and maybe backend forms?

But I have resolved the use of the generic `getSubmittedValue()` so should be ok now?